### PR TITLE
Start oss guide

### PR DESCRIPTION
**Pull request by LinuxBozo**
This tries to address https://www.pivotaltracker.com/story/show/176310462, providing guidance on how Truss approaches open source software. This allows us a space to maintain a canonical version of our Code Of Conduct.

While the Engineering Playbook covers lots of process, it doesn't specifically layout how and what we consider when doing open source. This tries to solve this, and eventually, provide more reasoning on why we do what we do, and the make the case that other orgs should strive to do the same.

### CODE REVIEW

File: opensource.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+-->
+
+# Community First
+One of the things that is imortant to us is making sure that we have a community around our projects that is safe for everyone to participate, no matter if they are fellow Trussels or not. This means that we have developed a strong [Code of Conduct](CODE-OF-CONDUCT) that governs how we manage the community and contributions to our projects. Our ultimate goal is to make sure that everyone has a voice and is able to contribute to our software in ways that are meaningful to them. You should always include a copy of the Code Of Conduct, or point to this canonical version so that the community at large is aware.
+
+# Licensing
+
+Documentation should be licensed as [CC-BY 4.0 International](LICENSE-CC-BY-4.0) when possible.
+
+Where it makes sense, source code should default to using a [BSD](LICENSE-BSD-3-CLAUSE) license.
```
Comment (ferlatte) [inline opensource.md:32]: This isn't correct. We have a TDR about licensing that should be followed for Truss related open source work: https://docs.google.com/document/d/12UYIN3XfRPdKQV87_0ILa9-z6eWRBcLqtwP9fCyR6Tg/edit

File: opensource.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- We do not have the rights to reproduce or publish the item on our own.
+- Publishing is restricted by law or other regulations, such as the Export Administration Regulations.
+
+<!--
+## Oh no, I've published an excepted case, what do?
+TODO
+-->
+
+# Community First
+One of the things that is imortant to us is making sure that we have a community around our projects that is safe for everyone to participate, no matter if they are fellow Trussels or not. This means that we have developed a strong [Code of Conduct](CODE-OF-CONDUCT) that governs how we manage the community and contributions to our projects. Our ultimate goal is to make sure that everyone has a voice and is able to contribute to our software in ways that are meaningful to them. You should always include a copy of the Code Of Conduct, or point to this canonical version so that the community at large is aware.
```
Comment (ferlatte) [inline opensource.md:26]: Typo. One of the things that is imortant -> "important"

File: opensource.md (hunk 1 above)
Comment (ferlatte) [reply opensource.md:32]: Not sure if you want to summarize, or link to the TDR from here.

**Review by ferlatte — CHANGES_REQUESTED** [review]

File: opensource.md (hunk 1 above)
Comment (LinuxBozo) [reply opensource.md:32]: Can do both. I hadn't seen that explicitly, and was going mostly off most of the terraform modules that have license files included that have been published. Assumptions get you ever time.

Comment (ferlatte) [reply opensource.md:32]: Heh. Yeah, I figured as much. For terraform, those should be licensed to fit that ecosystem which probably isn't Apache 2.0. Same with our React project(s). Turns out licensing is subtle and complex! :)

Comment (LinuxBozo) [reply opensource.md]: I _think_ I've addressed this properly now.

**Review by eeeady — APPROVED** [review]
LGTM. Thank you for adding this! I think we need a final go from @MichaelDBurstein or @ferlatte ?

**Review by ferlatte — APPROVED** [review]
This looks great. How exciting!

**Pull request merged**